### PR TITLE
[UnitaryHACK] Implemented E91 Protocol

### PR DESCRIPTION
**Pull request by pritamsinha2304**
Tried to implement the E91 protocol. So far the difficult one.

NOTE: I am trying these QKD protocols by learning, reading papers, watching videos and then trying it to implement it in code. So if there is some mistake in the code and also if there is mistake in my concept or I have understood the concept wrongly, please do tell me. I am also learning.

### CODE REVIEW

File: examples/QKD/e91_combined.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    result = bell_state(alice_key_decoy, bob_key_decoy.content)
+    print("Result Alice: {}".format(result))
+    if 2.0 <= abs(result) < 4:  # nearly equivalent to 2.82
+        print("|S| value: {}".format(abs(result)))
+    else:
+        print("Alice and Bob discard the bits and run again")
+
+
+def eve_sniffing_quantum(sender, receiver, qubit):
+    qubit.measure(non_destructive=True)
```
Comment (stephendiadamo) [inline examples/QKD/e91_combined.py:199]: Maybe something interesting is to measure only some of the qubits, like with some probability `p`, Eve measures a qubit, then you can see which values of `p` the protocol accepts the key.

**Review by stephendiadamo — COMMENTED** [review]
Looks good. I added a comment, let me know if you want to add that, or else I can merge the code.

File: examples/QKD/e91_combined.py (hunk 1 above)
Comment (pritamsinha2304) [reply examples/QKD/e91_combined.py:199]: Ok. Let me try. I will let you know if I can do it or else you can merge it.

Comment (pritamsinha2304) [reply examples/QKD/e91_combined.py:199]: Seems like I cannot do it. You can merge it. I won't add it.

**Review by stephendiadamo — APPROVED** [review]

**Pull request merged**